test/ComposableCoW.stoploss.t.sol

```diff
@@ -0,0 +1,151 @@
+        stopLoss.getTradeableOrder(safe, address(0), bytes32(0), abi.encode(data), bytes(""));
+        
+    }
+
+    function test_strikePriceNotMet_fuzz(int256 sellTokenOraclePrice, int256 buyTokenOraclePrice, int256 strike) public {
+        vm.assume(buyTokenOraclePrice > 0);
+        vm.assume(sellTokenOraclePrice > 0);
+        vm.assume(strike > 0);
+        vm.assume(sellTokenOraclePrice/buyTokenOraclePrice > strike);
+        
+        StopLoss.Data memory data = StopLoss.Data({
+            sellToken: SELL_TOKEN,
+            buyToken: BUY_TOKEN,
+            sellTokenPriceOracle: mockOracle(SELL_ORACLE, sellTokenOraclePrice),
+            buyTokenPriceOracle: mockOracle(BUY_ORACLE, buyTokenOraclePrice),
+            strike: strike,
+            sellAmount: 1 ether,
+            buyAmount: 1 ether,
+            appData: APP_DATA,
+            receiver: address(0x0),
+            isSellOrder: false,
+            isPartiallyFillable: false,
+            validtyBucketSeconds: 15 minutes
+        });
+        
+        vm.expectRevert(IConditionalOrder.OrderNotValid.selector);
+        stopLoss.getTradeableOrder(safe, address(0), bytes32(0), abi.encode(data), bytes(""));
+    }
+
+    function test_strikePriceMet_fuzz(int256 sellTokenOraclePrice, int256 buyTokenOraclePrice, int256 strike) public {
+        vm.assume(buyTokenOraclePrice > 0);
+        vm.assume(sellTokenOraclePrice > 0);
+        vm.assume(strike > 0);
+        vm.assume(sellTokenOraclePrice/buyTokenOraclePrice <= strike);
+        
+        StopLoss.Data memory data = StopLoss.Data({
+            sellToken: SELL_TOKEN,
+            buyToken: BUY_TOKEN,
+            sellTokenPriceOracle: mockOracle(SELL_ORACLE, sellTokenOraclePrice),
+            buyTokenPriceOracle: mockOracle(BUY_ORACLE, buyTokenOraclePrice),
+            strike: strike,
+            sellAmount: 1 ether,
+            buyAmount: 1 ether,
+            appData: APP_DATA,
+            receiver: address(0x0),
+            isSellOrder: false,
+            isPartiallyFillable: false,
+            validtyBucketSeconds: 15 minutes
+        });
+
+        // 25 June 2023 18:40:51
+        vm.warp(1687718451);
+
+        GPv2Order.Data memory order = stopLoss.getTradeableOrder(safe, address(0), bytes32(0), abi.encode(data), bytes(""));
+        assertEq(address(order.sellToken), address(SELL_TOKEN));
+        assertEq(address(order.buyToken), address(BUY_TOKEN));
+        assertEq(order.sellAmount, 1 ether);
+        assertEq(order.buyAmount, 1 ether);
+        assertEq(order.receiver, address(0x0));
+        assertEq(order.validTo, 1687718700);
+        assertEq(order.appData, APP_DATA);
+        assertEq(order.feeAmount, 0);
+        assertEq(order.kind, GPv2Order.KIND_BUY);
+        assertEq(order.partiallyFillable, false);
+        assertEq(order.sellTokenBalance, GPv2Order.BALANCE_ERC20);
+        assertEq(order.buyTokenBalance, GPv2Order.BALANCE_ERC20);
+    }
+
+    function test_validTo() public {
```

Comment:
This could potentially be migrated to a `fuzz` test for `validityBucketSeconds` to ensure that the parameter is passed and tested thoroughly.

Reply:
The issue I have with fuzz tests is that they basically reimplement the logic I have in the smart contract. How would you write the assertion of a fuzz test in a way that doesn't require writing the validTo computation formula again?

For now, I replaced the validity bucket here by another value than in the other tests which catches the issue you found.